rpc_component/schemata.py

```diff
@@ -134,20 +135,59 @@ def is_version_ids_unique(releases):
     }
 )
 
-component_metadata_schema = Schema(
+dependencies_schema = Schema(
     {
-        "dependencies": And(
+        Optional("dependencies"): And(
             [
                 {
                     "name": And(str, len),
                     "constraints": constraints_schema,
                 },
             ],
             is_value_unique("name"),
+        )
+    }
+)
+
+artifacts_file_schema = Schema(
+    {
+        "type": "file",
+        "source": And(str, len),
+        Optional("dest"): And(str, len),
+        Optional("expire_after"): And(str, len)
```

Comment:
I assume this must be in a specific format, you can make that part of the schema so that it is validated. For example, if it's a number of days you could use a function to check it's an integer greater than 0 or you could use a regex or maybe something from the `datetime` class.

Reply:
Good call, done.